File: programs/Frog - 2 Steps.cpp
#include <iostream>
#include <vector>
int main(void) {
    int n;
    std::cin >> n;
    std::vector<int> h(n), dp(std::max(n, 2));
    for (auto &it : h) std::cin >> it;

    dp[0] = 0;
    dp[1] = abs(h[1] - h[0]);

    for (int i = 2; i < n; ++i) {
        dp[i] = std::min(dp[i - 1] + abs(h[i] - h[i - 1]), dp[i - 2] + abs(h[i] - h[i - 2]));
    }

    std::cout << dp[n - 1] << "\n";

    return 0;
}

File: programs/Frog - 2 Steps.py
N = int(input())
height = list(map(int, input().split()))

min_cost = [None] * N
min_cost[0] = 0
min_cost[1] = abs(height[1] - height[0])
for i in range(2, N):
    min_cost[i] = min(min_cost[i-1] + abs(height[i] - height[i-1]),
                      min_cost[i-2] + abs(height[i] - height[i-2]))

print("Minimum cost is", min_cost[N - 1])
